python_ta/contracts/__init__.py

```diff
@@ -421,10 +440,18 @@ def wrapper(wrapped, instance, args, kwargs):
             klass_mod = _get_module(klass)
             if klass_mod is not None and ENABLE_CONTRACT_CHECKING:
                 _check_invariants(instance, klass, klass_mod.__dict__)
+
+                # Additionally check RI violations on PyTA-decorated instances that were mutated
+                # across the function call.
+                mutated_instances = getattr(instance_klass, "__mutated_instances__", [])
+                for mutable_instance in mutated_instances:
+                    _check_invariants(mutable_instance, klass, klass_mod.__dict__)
         except PyTAContractError as e:
             raise AssertionError(str(e)) from None
         else:
             return r
+        finally:
+            setattr(instance_klass, "__mutated_instances__", mutated_instances_to_restore)
```

Comment:
The restoring is generally good; however, we should have different cases for whether `__mutated_instances__` was an attribute or not originally. If it wasn't, we should use `delattr` to remove the instance attribute altogether, so it remains a temporary attribute.

Reply:
Sounds like a plan--added a branch for deleting the temporary attribute